kentikapi/cloud/export.go

```diff
@@ -184,78 +186,78 @@ type CloudExportStatus struct {
 type AzurePropertiesRequiredFields struct {
 	Location       string
 	ResourceGroup  string
 	StorageAccount string
 	SubscriptionID string
 }
 
-// CloudExportGCERequiredFields is subset of fields required to create a GCE CloudExport.
-type CloudExportGCERequiredFields struct {
+// GCEExportRequiredFields is subset of fields required to create a GCE Export.
+type GCEExportRequiredFields struct {
 	Name          string
 	PlanID        string
 	GCEProperties GCEPropertiesRequiredFields
 }
 
-// GCEPropertiesRequiredFields is subset of GCEProperties required to create a GCE CloudExport.
+// GCEPropertiesRequiredFields is subset of GCEProperties required to create a GCE Export.
 type GCEPropertiesRequiredFields struct {
 	Project      string
 	Subscription string
 }
 
-// CloudExportIBMRequiredFields is subset of fields required to create an IBM CloudExport.
-type CloudExportIBMRequiredFields struct {
+// IBMExportRequiredFields is subset of fields required to create an IBM Export.
+type IBMExportRequiredFields struct {
 	Name          string
 	PlanID        string
 	IBMProperties IBMPropertiesRequiredFields
 }
 
-// IBMPropertiesRequiredFields is subset of IBMProperties required to create an IBM CloudExport.
+// IBMPropertiesRequiredFields is subset of IBMProperties required to create an IBM Export.
 type IBMPropertiesRequiredFields struct {
 	Bucket string
 }
 
-// CloudExportType is the type of export task.
-type CloudExportType string
+// ExportType is the type of export task.
+type ExportType string
 
 const (
-	// CloudExportTypeUnspecified is invalid or incomplete cloud export.
-	CloudExportTypeUnspecified CloudExportType = "CLOUD_EXPORT_TYPE_UNSPECIFIED"
-	// CloudExportTypeKentikManaged is for cloud exports that are managed by Kentik.
-	CloudExportTypeKentikManaged CloudExportType = "CLOUD_EXPORT_TYPE_KENTIK_MANAGED"
-	// CloudExportTypeCustomerManaged is for cloud exports that are managed by Kentik customers,
+	// ExportTypeUnspecified is invalid or incomplete cloud export.
+	ExportTypeUnspecified ExportType = "CLOUD_EXPORT_TYPE_UNSPECIFIED"
+	// ExportTypeKentikManaged is for cloud exports that are managed by Kentik.
+	ExportTypeKentikManaged ExportType = "CLOUD_EXPORT_TYPE_KENTIK_MANAGED"
+	// ExportTypeCustomerManaged is for cloud exports that are managed by Kentik customers,
 	// e.g. by running an agent.
-	CloudExportTypeCustomerManaged CloudExportType = "CLOUD_EXPORT_TYPE_CUSTOMER_MANAGED"
+	ExportTypeCustomerManaged ExportType = "CLOUD_EXPORT_TYPE_CUSTOMER_MANAGED"
 )
 
-// CloudProvider is the name of cloud provider.
-type CloudProvider string
+// Provider is the name of cloud provider.
+type Provider string
 
 const (
-	CloudProviderAlibaba CloudProvider = "alibaba"
-	CloudProviderAWS     CloudProvider = "aws"
-	CloudProviderAzure   CloudProvider = "azure"
-	CloudProviderGCE     CloudProvider = "gce" // gcp value in Agents API
-	CloudProviderIBM     CloudProvider = "ibm"
+	ProviderAlibaba Provider = "alibaba"
```

Comment:
(unrelated to the change in this review) I think we can drop Alibaba. We do not support it yet and it is unclear if/when we will.

Reply:
I'll remove it in next PR